### DISCUSSION TITLE
filter updates

### DISCUSSION THREAD

**Discussion by ahoyahoy** [ueberdosis/hocuspocus#333] [Q&A]
Is it possible to filter incoming updates (server) for forbidden fields?
When I have a shared document for users with granulated permissions and I need to prevent them from passing or getting fields outside of their permissions.

**Comment by georeith** [comment]
Attempting to do this within a single CRDT document feels wrong as the whole purpose of the CRDT is to acheive eventual consistency and what you're asking for (filtering updates within a document) is some sort of selective perpetual divergence.

I think utilising YJS subdocuments (or separate documents) to store those fields and applying auth to those subdocuments (preventing those without permission to load the subdocument at all) would achieve this without trying to add selective divergence to the CRDT.
